### PR TITLE
[DESKTOP/API] Register & Login flow

### LINKED ISSUE #2: Integrate Log In interface
The interface should be divided into **two parts**:
  - left part: log in form with "remember me" checkbox
  - right part: 3 tabs (blog, docs, support)

### Tasks
- Front-End
  - [x] Left part
  - [ ] Right part
  - [x] Comments
  - [x] Unit tests
- Back-End
  - [x] JWT token-based authentication
  - [x] isAuthenticated middleware
  - [x] Unique email address
  - [x] Comments
  - [x] Unit tests

### LINKED ISSUE #27: Integrate Register interface
Need #26 to be done before.

### PR DESCRIPTION
**Pull request by jvnm-dev**
# PR Details

Finishing api bootstrapping & register flow

Closes #2 
Closes #27 

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] My code is commented
- [ ] All new and existing tests passed.

### CODE REVIEW

File: projects/api/src/controllers/authController.ts [hunk 1]
```diff
@@ -1,12 +1,67 @@
...
+
+    const passwordMatch = await bcrypt.compare(password, user.password)
+
+    if (!passwordMatch) {
+        return Boom.badRequest('incorrect_password')
+    }
+
+    delete user.password
+
+    const today = DateTime.local().toSeconds()
```
Comment (Sylv11) [inline projects/api/src/controllers/authController.ts:31]: It is better to use the moment library for dates and moments

File: projects/api/src/controllers/authController.ts [hunk 2]
```diff
@@ -1,12 +1,67 @@
...
+    const dayInSeconds = 86400
+
+    return jwt.sign({
+        ...user,
+        exp: remember ? today + (30 * dayInSeconds): today + dayInSeconds,
+    }, secretKey)
+}
```
Comment (Sylv11) [inline projects/api/src/controllers/authController.ts:38]: Can you explain me

File: projects/api/src/controllers/authController.ts [hunk 3]
```diff
@@ -1,12 +1,67 @@
...
+        // Check unique email
+        const emailTaken = await repository.count({ where: { email: data.email }})
+
+        if (emailTaken) {
+            return Boom.badRequest('email_already_taken')
+        }
+
+        // Hashing password
+        const saltRounds: number = parseInt(`${process.env.SALT_ROUNDS}`, 10)
+        const clearPassword = data.password
```
Comment (Sylv11) [inline projects/api/src/controllers/authController.ts]: Type ?

File: projects/desktop/src/components/RegisterScreen/index.tsx [hunk 4]
```diff
@@ -1,29 +1,103 @@
...
+import { Box, Logo, RegisterLanguageChooser } from './styles'
 import logo from '../../assets/img/logo-small.png'
-import { Form as FinalForm } from 'react-final-form'
-import {Form} from './form'
+import { Form as FinalForm, AnyObject } from 'react-final-form'
+import { Form } from './form'
+import axios, { AxiosResponse } from 'axios'
+import { serverURL } from '../../constants'
+import _ from 'lodash'
+
```
Comment (Sylv11) [inline projects/desktop/src/components/RegisterScreen/index.tsx:12]: By the way, for all the imports, I think it is better to group the external libraries and then a blank line and then all the internal imports

File: projects/api/src/controllers/authController.ts (hunk 2 above)
Comment (jvnm-dev) [reply projects/api/src/controllers/authController.ts:38]: Exp(iration) must be seconds. For non-remember sessions I put 1 day, for the remember sessions I put a month

**Pull request merged**